model_compression_toolkit/core/common/substitutions/batchnorm_refusing.py

```diff
@@ -0,0 +1,186 @@
+            epsilon_str: The framework specific attribute name of the batch norm layer's epsilon parameter.
+            use_bias: The framework specific attribute name of the convolution layer's bias flag.
+            layer_name_str: The framework specific attribute name of layer's name.
+        """
+        super().__init__(matcher_instance=EdgeMatcher(source_node, bn_node))
+        self.update_kernel_for_bn_refusing_fn = update_kernel_for_bn_refusing_fn
+        self.kernel_str = kernel_str
+        self.bias_str = bias_str
+        self.gamma_str = gamma_str
+        self.beta_str = beta_str
+        self.moving_mean_str = moving_mean_str
+        self.moving__variance_str = moving__variance_str
+        self.epsilon_str = epsilon_str
+        self.use_bias = use_bias
+        self.layer_name_str = layer_name_str
+
+    def substitute(self,
+                   graph: Graph,
+                   edge_nodes: Tuple[BaseNode, BaseNode]) -> Graph:
+        """
+        Re-fuse BatchNormalization into preceding linear layers.
+
+        Args:
+            graph: Graph we apply the substitution on.
+            edge_nodes: Tuple of tow nodes (linear op and batchnorm node).
+
+        Returns:
+            Graph after applying the substitution.
+        """
+
+        num_nodes_before_substitution = len(graph.nodes)
+        num_edges_before_substitution = len(graph.edges)
+
+        conv_node = edge_nodes[0]
+
+        # If the linear operator is part of a reused group (it is the "base" node, or a reused node),
+        # we should skip the substitution.
+        if conv_node.reuse or conv_node.reuse_group is not None:
+            return graph
+
+        bn_node = edge_nodes[1]
+
+        if len(graph.get_next_nodes(conv_node)) > 1 or len(graph.get_prev_nodes(bn_node)) > 1:
+            return graph
+
+        kernel = conv_node.get_weights_by_keys(self.kernel_str)
+        bias = conv_node.get_weights_by_keys(self.bias_str)
+        gamma = bn_node.get_weights_by_keys(self.gamma_str)
+        beta = bn_node.get_weights_by_keys(self.beta_str)
+        moving_mean = bn_node.get_weights_by_keys(self.moving_mean_str)
+        moving_variance = bn_node.get_weights_by_keys(self.moving__variance_str)
+        eps = bn_node.framework_attr[self.epsilon_str]
+
+        # TODO:
+        # if gamma or beta is None raise Exception
+        if gamma is None:
+            gamma = 1.0
+        if beta is None:
+            beta = 0.0
+
+        if bias is None:
+            bias = 0.0
+
+        weights_scale = gamma / np.sqrt(moving_variance + eps)
+        bias = beta + (bias - moving_mean) * weights_scale
+
+        kernel, kernel_name = self.update_kernel_for_bn_refusing_fn(conv_node, kernel, weights_scale)
+
+        framework_attr = copy.copy(conv_node.framework_attr)
+        framework_attr[self.use_bias] = True
+        if self.layer_name_str is not None:
+            framework_attr[self.layer_name_str] = conv_node.name + '_refused'
```

Comment:
Please use it as a constant (not in constants file, as this is relevant only to this scope, if I'm not missing anything)